src/tslint.ts

```diff
@@ -99,13 +98,19 @@ class Linter {
         return rules.some((r) => r.equals(rule));
     }
 
-    private computeFullOptions() {
-        let {configuration, rulesDirectory} = this.options;
-        if (configuration == null) {
-            configuration = DEFAULT_CONFIG;
+    private computeFullOptions(options: ILinterOptionsRaw): ILinterOptions {
+        if (!options || typeof options !== "object") {
+            options = {};
         }
-        this.options.rulesDirectory = arrayify(rulesDirectory).concat(arrayify(configuration.rulesDirectory));
-        this.options.configuration = configuration;
+
+        let { configuration, formatter, formattersDirectory, rulesDirectory } = options;
+
+        return {
+            configuration: configuration || DEFAULT_CONFIG,
+            formatter: formatter || "json",
```

Comment:
let's just have it default to "prose", since that's what the CLI does. The results are still accessible programmatically alongside of the prose description